core/corehttp/gateway_handler.go

```diff
@@ -290,6 +295,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		return
```

Comment:
Technically, if a 404 page is present, we should return 404 here. But I'm fine leaving that edge-case unfixed.